post_install_node_config.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from kubernetes import client, config
+from openshift.dynamic import DynamicClient
+import sys
+import argparse
+import subprocess
+
+# Make aws related config changes such as security group rules etc
+def _aws_config(nodes,clustername):
+    vpc_cmd = ["aws ec2 describe-instances --query 'Reservations[*].Instances[*].[InstanceId,Tags[?Key==`Name`].Value|[0],State.Name,PrivateIpAddress,PublicIpAddress, PrivateDnsName, VpcId]' --output text | column -t | grep " + clustername + "| awk '{print $7}' | grep -v '^$' | sort -u"]
+    process = subprocess.Popen(vpc_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    stdout,stderr = process.communicate()
+    cluster_vpc = stdout.decode("utf-8")
+    cluster_vpc = cluster_vpc.replace('\n','')
+    cluster_vpc = cluster_vpc.replace(' ','')
+    
+    sec_grp_cmd = ["aws ec2 describe-security-groups --filters \"Name=vpc-id,Values=" + cluster_vpc + "\" --output json | jq .SecurityGroups[].GroupId"]
+    process = subprocess.Popen(sec_grp_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    stdout,stderr = process.communicate()
+    sec_group = stdout.decode("utf-8")
+
+    sec_group = sec_group.replace(' ','')
+    sec_group = sec_group.replace('"','')
+    sec_group = sec_group.replace('\n',' ')
+    sec_group_list = list(sec_group.split(" "))
+
+    sec_group_cmds = [ "aws ec2 authorize-security-group-ingress --group-id ITEM --protocol tcp --port 22 --cidr 0.0.0.0/0",
+                       "aws ec2 authorize-security-group-ingress --group-id ITEM --protocol tcp --port 2022 --cidr 0.0.0.0/0",
+                       "aws ec2 authorize-security-group-ingress --group-id ITEM --protocol tcp --port 20000-20109 --cidr 0.0.0.0/0",
+                       "aws ec2 authorize-security-group-ingress --group-id ITEM --protocol udp --port 20000-20109 --cidr 0.0.0.0/0",
+                       "aws ec2 authorize-security-group-ingress --group-id ITEM --protocol tcp --port 32768-60999 --cidr 0.0.0.0/0",
+                       "aws ec2 authorize-security-group-ingress --group-id ITEM --protocol udp --port 32768-60999 --cidr 0.0.0.0/0"]
+    for item in sec_group_list:
+        if item == "":
+            return
+        else:
+            for cmnd in sec_group_cmds:
+                my_cmd = cmnd.replace('ITEM',item)
```

Comment:
this is causing lint error, it seems that my_cmd is never used

Line below, the subprocess is using sec_grp_cmd, should it be my_cmd?

Reply:
good catch resolved.